packages/flutter/lib/src/material/app.dart

```diff
@@ -52,6 +52,7 @@ class MaterialApp extends StatefulWidget {
     this.theme,
     this.home,
     this.routes: const <String, WidgetBuilder>{},
+    this.initialRoute,
```

Comment:
MaterialApp is supposed to be a one-stop shop for initializing the root of your app.  If we add a Navigator argument, we'll make every app more complicated.  I agree that it could be a problem if we spam too many properties onto this object...  I'm not sure what else to do besides taking one more step down this path.